src/node.h

```diff
@@ -700,69 +700,38 @@ v8::MaybeLocal<v8::Value> MakeCallback(v8::Isolate* isolate,
 /* Helper class users can optionally inherit from. If
  * `AsyncResource::MakeCallback()` is used, then all four callbacks will be
  * called automatically. */
-class AsyncResource {
+class NODE_EXTERN AsyncResource {
  public:
   AsyncResource(v8::Isolate* isolate,
                 v8::Local<v8::Object> resource,
                 const char* name,
-                async_id trigger_async_id = -1)
-      : isolate_(isolate),
-        resource_(isolate, resource) {
-    async_context_ = EmitAsyncInit(isolate, resource, name,
-                                   trigger_async_id);
-  }
+                async_id trigger_async_id = -1);
 
-  virtual ~AsyncResource() {
-    EmitAsyncDestroy(isolate_, async_context_);
-    resource_.Reset();
-  }
+  virtual ~AsyncResource();
```

Comment:
https://docs.microsoft.com/en-us/cpp/cpp/dllexport-dllimport?view=vs-2017#remarks
```
If a class is marked declspec(dllexport), any specializations of class templates in the class hierarchy are implicitly marked as declspec(dllexport). This means that class templates are explicitly instantiated and the class's members must be defined.
```
IIUC the for `NODE_EXTERN` the compiler explicitly instantiates all members and defaults, specifically the copy ctor and copy operator. So the tradeoff for making this `NODE_EXTERN` is:

```suggestion
  AsyncResource operator=(const AsyncResource&) = delete;
  AsyncResource(const AsyncResource&) = delete;

  virtual ~AsyncResource();
```